inference-engine/thirdparty/clDNN/api/cldnn/runtime/debug_configuration.hpp

```diff
@@ -0,0 +1,27 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+#include <string>
+#include <mutex>
+
+#ifdef GPU_DEBUG_CONFIG
+#define GPU_DEBUG_IF(cond) if (cond)
+#else
+#define GPU_DEBUG_IF(cond) if (0)
+#endif
+
+namespace cldnn {
+
+class debug_configuration {
+private:
+    debug_configuration();
+public:
+    static const char *prefix;
+    int verbose;
+    std::string dump_graphs;
+    static const debug_configuration *get_instance();
+};
+
```

Comment:
Might this singleton cause crash when it is used from multi threads? e.g., first thread created and destroyed this but second thread is still alive and trying to access this.. Maybe we need to manage reference count and delete it when the reference count is 0?

Reply:
To my knowledge, static is similar to global, which means that it is not destroyed though class instance or thread is destroyed.

Reply:
BTW, not sure whether all compilers are doing as I expect ;) In that case, the alternative would be to make it as global static variable..

Reply:
Ah you are correct, I was being confused other use case where we had to release in some point that the object is not being used for saving resource. This is not the case.